### PR TITLE
Enable conv2d operator for a ngraph engine

### PR DESCRIPTION
**Pull request by mozga-intel**
Introduces the nGraph's implementation of the conv2d operator.

### CODE REVIEW

**Comment by mozga-intel** [comment]
@luotao1, @tensor-tang Can you please review and comment this pull-request?

File: paddle/fluid/framework/ngraph_operator.cc [hunk 1]
```diff
@@ -302,21 +302,24 @@ void NgraphEngine::GetNgInputShape(std::shared_ptr<OperatorBase> op) {
...
-      auto* var = scope_.FindVar(var_name);
-      if (var && var->IsType<LoDTensor>()) {
-        auto* tensor_pd = GetLoDTensorOrSelectedRowsValueFromVar(*var);
-        auto& ddim = tensor_pd->dims();
-        auto ng_shape = Ddim2Shape(ddim);
-        auto ng_type = var_type_map_.at(var_name);
-        auto prm =
-            std::make_shared<ngraph::op::Parameter>(ng_type, ng_shape, true);
-        (*var_node_map_)[var_name] = prm;
+  for (auto& op : fused_ops_) {
```
Comment (luotao1) [inline paddle/fluid/framework/ngraph_operator.cc]: why do you update here for adding conv2d_op?

Comment (mozga-intel) [reply paddle/fluid/framework/ngraph_operator.cc]: The following list had not being completed as of yet - basically is ancient as the hill. To alleviate this utterance, I want to say that the list of the output tensors is completed right now. Notably, We're having these changes added to other pull-request.

Comment (mozga-intel) [reply paddle/fluid/framework/ngraph_operator.cc]: The following list had not been completed as of yet - basically it is as ancient as the hill. To improve it, I want to say that the list of the output tensors is completed right now. Thus, we will be able to get some information about the output tensors, i.e the size of the tensors. Also, we are going to have these changes added to other pull-request.

**Comment by mozga-intel** [comment]
@luotao1 Can you please continue review or merge it?

File: paddle/fluid/framework/ngraph_operator.cc (hunk 1 above)
Comment (baojun-nervana) [reply paddle/fluid/framework/ngraph_operator.cc]: @mozga-intel This change is no longer needed as PR #14948 merged. The ngraph_operator.cc is pending to be removed per PR #15537. Also can you address the conflict from PR #15538?

Comment (mozga-intel) [reply paddle/fluid/framework/ngraph_operator.cc]: @baojun-nervana Great!, I just saw, I will take these changes into account to resolve these conflicts.

File: paddle/fluid/operators/ngraph/ops/conv2d_op.h [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+#include <string>
+#include <vector>
+#include "ngraph/ngraph.hpp"
+#include "paddle/fluid/platform/ngraph_helper.h"
+
+namespace paddle {
+namespace operators {
+namespace ngraphs {
+
+std::shared_ptr<ngraph::Node> GroupedConvolution(
```
Comment (tensor-tang) [inline paddle/fluid/operators/ngraph/ops/conv2d_op.h:26]: @baojun-nervana @mozga-intel  I saw many `shared_ptr` used in ngraph, do you have any plan to replace with `unique_ptr`.

Comment (baojun-nervana) [reply paddle/fluid/operators/ngraph/ops/conv2d_op.h:26]: Let's look into this and address this in a separate PR.

Comment (tensor-tang) [reply paddle/fluid/operators/ngraph/ops/conv2d_op.h:26]: Sure

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**